DependencyInjection/Compiler/MaybeSetMimeServicesAsAliasesCompilerPass.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * Changes the `liip_imagine.mime_type_guesser` and `liip_imagine.extension_guesser` services to be aliases of the
+ * `mime_types` service provided by the FrameworkBundle when available.
+ *
+ * This compiler pass can be removed when dropping support for Symfony 4.2 and earlier.
+ *
+ * @internal
+ */
+final class MaybeSetMimeServicesAsAliasesCompilerPass extends AbstractCompilerPass
```

Comment:
Purposefully final and internal to make it clear this is a B/C layer that can (and should) go away when older Symfony version support is dropped.